### ISSUE TITLE
Freeze in FactArg

### ISSUE THREAD

**Issue by tueda** [vermaseren/form#6]
In the following code, FactArg seems never to return. (At least extremely slow.)

```
CF den;
S ep,w;
L F =
   + den(226492416*ep^4*w - 679477248*ep^4*w^2 + 452984832*ep^4*
  w^3 + 452984832*ep^4*w^4 - 679477248*ep^4*w^5 + 226492416*ep^4*
  w^6 - 1887436800*ep^5*w + 5775556608*ep^5*w^2 - 4001366016*ep^5
  *w^3 - 3774873600*ep^5*w^4 + 5888802816*ep^5*w^5 - 2000683008*
  ep^5*w^6 - 1730150400*ep^6*w + 4756340736*ep^6*w^2 - 2592079872
  *ep^6*w^3 - 3460300800*ep^6*w^4 + 4322230272*ep^6*w^5 -
  1296039936*ep^6*w^6 + 51111788544*ep^7*w - 156153937920*ep^7*
  w^2 + 107860721664*ep^7*w^3 + 102223577088*ep^7*w^4 -
  158972510208*ep^7*w^5 + 53930360832*ep^7*w^6 - 88394956800*ep^8
  *w + 278057189376*ep^8*w^2 - 202534551552*ep^8*w^3 -
  176789913600*ep^8*w^4 + 290929508352*ep^8*w^5 - 101267275776*
  ep^8*w^6 - 350459265024*ep^9*w + 1065105752064*ep^9*w^2 -
  728374444032*ep^9*w^3 - 700918530048*ep^9*w^4 + 1078833709056*
  ep^9*w^5 - 364187222016*ep^9*w^6 + 1143994318848*ep^10*w -
  3545436782592*ep^10*w^2 + 2514896289792*ep^10*w^3 +
  2287988637696*ep^10*w^4 - 3658890608640*ep^10*w^5 +
  1257448144896*ep^10*w^6 + 54207184896*ep^11*w - 101166612480*
  ep^11*w^2 - 14495514624*ep^11*w^3 + 108414369792*ep^11*w^4 -
  39711670272*ep^11*w^5 - 7247757312*ep^11*w^6 - 3522057732096*
  ep^12*w + 10869824028672*ep^12*w^2 - 7651417128960*ep^12*w^3 -
  7044115464192*ep^12*w^4 + 11173474861056*ep^12*w^5 -
  3825708564480*ep^12*w^6 + 3387722563584*ep^13*w -
  10557767811072*ep^13*w^2 + 7564645367808*ep^13*w^3 +
  6775445127168*ep^13*w^4 - 10952367931392*ep^13*w^5 +
  3782322683904*ep^13*w^6 + 1380798431232*ep^14*w - 4224234553344
  *ep^14*w^2 + 2925275381760*ep^14*w^3 + 2761596862464*ep^14*w^4
   - 4306073812992*ep^14*w^5 + 1462637690880*ep^14*w^6 -
  3140694835200*ep^15*w + 9744207052800*ep^15*w^2 - 6925634764800
  *ep^15*w^3 - 6281389670400*ep^15*w^4 + 10066329600000*ep^15*w^5
   - 3462817382400*ep^15*w^6 + 1087163596800*ep^16*w -
  3382286745600*ep^16*w^2 + 2415919104000*ep^16*w^3 +
  2174327193600*ep^16*w^4 - 3503082700800*ep^16*w^5 +
  1207959552000*ep^16*w^6)
;
factarg den;
.end
```

**Comment by benruijl** [comment]
This is indeed a bug in the Hensel lifting. The problem has to do with the leading coefficients becoming bad during the lifting (apparently this is a side effect). Therefore, they have to be reset to the correct values. The lifting algorithm consists of a simple case, which is called when the number of terms is not larger than POLYFACT_MAX_PREDETERMINATION and the generic case. This last one has a bug, so that's why you don't see the bug if POLYFACT_MAX_PREDETERMINATION is large.

I will push a patch that fixes the problem, though I am not sure if this fix works in all possible scenarios. Therefore, I will leave this bug open for now.

**Comment by jodavies** [comment]
@benruijl 's fix was 6f06e94 . After all these years it feels safe to close this one.

**Comment by tueda** [comment]
Thank you for pointing it out. I close this issue.
